src/vm/hints/uint256_utils.rs

```diff
@@ -0,0 +1,519 @@
+use crate::types::relocatable::MaybeRelocatable;
+use crate::vm::errors::vm_errors::VirtualMachineError;
+use crate::vm::hints::hint_utils::{
+    get_address_from_var_name, get_integer_from_address_plus_offset, get_integer_from_var_name,
+};
+use crate::vm::vm_core::VirtualMachine;
+use num_bigint::BigInt;
+use num_traits::FromPrimitive;
+use std::collections::HashMap;
+use std::ops::{Shl, Shr};
+
+/*
+Implements hint:
+%{
+    sum_low = ids.a.low + ids.b.low
+    ids.carry_low = 1 if sum_low >= ids.SHIFT else 0
+    sum_high = ids.a.high + ids.b.high + ids.carry_low
+    ids.carry_high = 1 if sum_high >= ids.SHIFT else 0
+%}
+*/
+pub fn uint256_add(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    let shift: BigInt = bigint!(2).pow(128);
+
+    let a_addr = get_address_from_var_name("a", ids.clone(), vm, hint_ap_tracking)?;
+    let b_addr = get_address_from_var_name("b", ids.clone(), vm, hint_ap_tracking)?;
+    let carry_high_addr =
+        get_address_from_var_name("carry_high", ids.clone(), vm, hint_ap_tracking)?;
+    let carry_low_addr = get_address_from_var_name("carry_low", ids, vm, hint_ap_tracking)?;
+
+    let a_low = get_integer_from_address_plus_offset(&a_addr, 0, vm)?;
+    let a_high = get_integer_from_address_plus_offset(&a_addr, 1, vm)?;
+    let b_low = get_integer_from_address_plus_offset(&b_addr, 0, vm)?;
+    let b_high = get_integer_from_address_plus_offset(&b_addr, 1, vm)?;
+
+    // Hint main logic
+    // sum_low = ids.a.low + ids.b.low
+    // ids.carry_low = 1 if sum_low >= ids.SHIFT else 0
+    // sum_high = ids.a.high + ids.b.high + ids.carry_low
+    // ids.carry_high = 1 if sum_high >= ids.SHIFT else 0
+
+    let carry_low = if a_low + b_low >= shift {
+        bigint!(1)
+    } else {
+        bigint!(0)
+    };
+
+    let carry_high = if a_high + b_high + carry_low.clone() >= shift {
+        bigint!(1)
+    } else {
+        bigint!(0)
+    };
+
+    match (
+        vm.memory
+            .insert(&carry_high_addr, &MaybeRelocatable::from(carry_high)),
+        vm.memory
+            .insert(&carry_low_addr, &MaybeRelocatable::from(carry_low)),
+    ) {
+        (Ok(_), Ok(_)) => Ok(()),
+        (Err(error), _) | (_, Err(error)) => Err(VirtualMachineError::MemoryError(error)),
+    }
+}
+
+/*
+Implements hint:
+    %{
+        ids.low = ids.a & ((1<<64) - 1)
+        ids.high = ids.a >> 64
+    %}
+*/
+pub fn split_64(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    let a = get_integer_from_var_name("a", ids.clone(), vm, hint_ap_tracking)?;
+    let high_addr = get_address_from_var_name("high", ids.clone(), vm, hint_ap_tracking)?;
+    let low_addr = get_address_from_var_name("low", ids, vm, hint_ap_tracking)?;
+
+    let low: BigInt = a & (bigint!(1).shl(64_usize) - 1);
```

Comment:
```suggestion
    let low: BigInt = a & bigint!(u64::MAX);
```

Reply:
This would be even better (no intermediate allocations), but it seems the library authors didn't think of it:
```suggestion
    let low: BigInt = a & u64::MAX;
```